algorithms/src/std_algorithms/Kokkos_FindEnd.hpp

```diff
@@ -24,70 +24,89 @@
 namespace Kokkos {
 namespace Experimental {
 
+//
+// overload set accepting execution space
+//
+
 // overload set 1: no binary predicate passed
-template <class ExecutionSpace, class IteratorType1, class IteratorType2>
+template <
+    typename ExecutionSpace, typename IteratorType1, typename IteratorType2,
+    std::enable_if_t<::Kokkos::is_execution_space_v<ExecutionSpace>, int> = 0>
```

Comment:
Is there a reason these guys are not constrained on are_iterators like the equal algorithm?

Reply:
the different # of params does not cause the same ambiguous error from compiler